File: CanzoniereWPF/Data/IDataService.vb
Imports System.Collections.ObjectModel

Public Interface IDataService

    Function GetCurrentFolderAsync() As Task(Of Uri)
    Function GetCanzoneFromFilePathAsync(filePath As String) As Task(Of Canzone)
    Function GetFileElementsFromFolderAsync(folderPath As String) As Task(Of List(Of FileElement))

    Function CreateIndexAsync(folderPath As String) As Task
End Interface

File: KantaroWpf/Services/IDataService.cs
using Enzo.Music.KantaroWpf.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace Enzo.Music.KantaroWpf.Services;

public interface IDataService
{
    Task<string> GetCurrentFolderAsync();
    Task<Canzone?> GetCanzoneFromFilePathAsync(string filePath);
    Task<List<FileElement>> GetFileElementsFromFolderAsync(string folderPath);
    Task CreateIndexAsync(string folderPath);
}
